parser-typechecker/src/Unison/Lexer.hs

```diff
@@ -21,11 +29,38 @@ import           GHC.Exts (sortWith)
 import           Text.Megaparsec.Error (ShowToken(..))
 import           Unison.ShortHash ( ShortHash )
 import qualified Unison.ShortHash as SH
+import qualified Text.Megaparsec as P
+import qualified Text.Megaparsec.Error as EP
+import qualified Text.Megaparsec.Char as CP
+import Text.Megaparsec.Char (char)
+import qualified Text.Megaparsec.Char.Lexer as LP
+import qualified Unison.Util.Bytes as Bytes
+
+type Line = Int
+type Column = Int
+data Pos = Pos {-# Unpack #-} !Line {-# Unpack #-} !Column deriving (Eq,Ord,Show)
+type BlockName = String
+type Layout = [(BlockName,Column)]
+
+data Token a = Token {
+  payload :: a,
+  start :: !Pos,
+  end :: !Pos
+} deriving (Eq, Ord, Show, Functor)
+
+data ParsingEnv =
+  ParsingEnv { layout :: !Layout -- layout stack
+             , opening :: Maybe String } -- `Just b` if a block of type `b` is being opened
```

Comment:
Comment mentions a type var that looks like it was monomorphized to `String` afterwards.

Reply:
Should that `String` be `BlockName`?

Reply:
Good catch, yes.